lib/foreman_discovery/engine.rb

```diff
@@ -139,6 +139,11 @@ class Engine < ::Rails::Engine
           :parent => :configure_menu,
           :after=> :hostgroups
 
+        # strong parameters
+        parameter_filter Subnet, :discovery_id
+        parameter_filter Host::Managed, :discovery_rule_id
+        parameter_filter Hostgroup, :discovery_rules, :discovery_rule_ids, :discovery_rule_names, :type
```

Comment:
Should be `:type, :discovery_rules => [], :discovery_rule_ids => [], :discovery_rule_names => []` to permit an array of IDs/names.